### PR TITLE
recursive search for migrations within specific folders

### LINKED ISSUE #3: recursive search for migrations within specific folders
Hellow, when I am trying to use your gem, then not all migration files are found within the folder `db/migrate`
I have the same subfolders as 1.0, 2.0 e.t.c.

I would like to search for migration files not only inside `db/migrate/*` but do a recursive search for migrations within this folder.
Something like that `db/migrate/**/*`.

### PR DESCRIPTION
**Pull request by Ivanov-Anton**
closes #3

Do a recursive search for migrations within specific folders.
Instead of `db/migrate/*` prefer to use `db/migrate/**/*` pattern

### CODE REVIEW

**Comment by Ivanov-Anton** [comment]
@rwojsznis Hello, can you see my suggestion, please? I suggest recursively searching for migrations.

**Comment by rwojsznis** [comment]
Thanks again for your contribution @Ivanov-Anton! Before we proceed with code review can you please:
- rebase with `main` (so we don't test on Ruby 3.1 just yet)
- ensure introduced change doesn't break existing functionality
- add tests for newly introduced change

Thanks!

**Comment by rwojsznis** [comment]
Thank you for the follow-up! I'm pretty occupied atm, but I will try to review this within next ~7 days!

(worse case scenario will merge it and potentially slightly rework it according to some of my suggestions if you won't mind - I'm aware some of open source maintainers do that and I think it shortens the waiting time for the new release, a reasonable bargain I suppose when both parties have other things going on 👀 )

**Comment by Ivanov-Anton** [comment]
> Thank you for the follow-up! I'm pretty occupied atm, but I will try to review this within next ~7 days!
> 
> (worse case scenario will merge it and potentially slightly rework it according to some of my suggestions if you won't mind - I'm aware some of open source maintainers do that and I think it shortens the waiting time for the new release, a reasonable bargain I suppose when both parties have other things going on 👀 )

OK

File: lib/conflict_free_schema/database/schema_version_files.rb [hunk 1]
```diff
@@ -57,10 +57,15 @@ def self.init_schema_directory
...
         migration_directories.each_with_object([]) do |directory, migration_versions|
-          directory_migrations = Dir.glob(MIGRATION_VERSION_GLOB, base: directory)
-          directory_versions = directory_migrations.map! { |m| m.split('_').first }
+          directory_migrations = Dir.glob(directory)
+          directory_versions = directory_migrations.map do |migration|
+            item = Pathname.new(migration)
+            next if item.directory?
 
-          migration_versions.concat(directory_versions)
+            item.basename.to_s.match(/20[0-9][0-9]*/).to_s
```
Comment (rwojsznis) [inline lib/conflict_free_schema/database/schema_version_files.rb:65]: seems now we have handling `MIGRATION_VERSION_GLOB` constant 🤔 will take care of that post-merge tho!

File: lib/conflict_free_schema/database/schema_version_files.rb [hunk 2]
```diff
@@ -57,10 +57,15 @@ def self.init_schema_directory
...
+          end
+
+          migration_versions.concat(directory_versions.delete_if(&:blank?))
```
Comment (rwojsznis) [inline lib/conflict_free_schema/database/schema_version_files.rb:68]: we probably can just use `compact` on `L66` to simplify it a little bit

File: spec/conflict_free_schema/database/schema_version_files_spec.rb [hunk 3]
```diff
@@ -54,6 +56,61 @@
         end
       end
     end
+
+    context 'when "db/migrate" folder contains other subfolders' do
```
Comment (rwojsznis) [inline spec/conflict_free_schema/database/schema_version_files_spec.rb:60]: as those tests were stolen (d'oh) from gitlab directly - I think that maybe I will extract new unit tests to a separate spec file

Ideally this lib should be updated according [to this suggestion](https://github.com/stolen-ruby/conflict_free_schema/issues/2#issuecomment-875130221) so we can have support for multiple databases - keeping unit tests in sync (with gitlab) will probably make upgrade process smoother 🤔

**Review by rwojsznis — APPROVED** [review]
Nice 👏  thanks for the contribution, will merge & do some post-cleanups this week I hope (if you're avail feel free to take a look at my suggestions tho!)

File: spec/conflict_free_schema/database/schema_version_files_spec.rb (hunk 3 above)
Comment (Ivanov-Anton) [reply spec/conflict_free_schema/database/schema_version_files_spec.rb:60]: At the moment in the GitLab repo "Conflict-free schema version handling" feature does not support the recursive search for migrations within `db/migration`.

So this library will be slightly better than the implementation in GitLab.

Comment (rwojsznis) [reply spec/conflict_free_schema/database/schema_version_files_spec.rb:60]: I suppose it depends on one's needs 😉 

Thanks again for the contribution, will merge and refer to rest of my comments directly on main branch

**Pull request merged**